micro_http/src/request.rs

```diff
@@ -125,13 +129,26 @@ impl<'a> RequestLine<'a> {
 /// Wrapper over an HTTP Request.
 #[allow(unused)]
 #[derive(Debug)]
-pub struct Request<'a> {
-    request_line: RequestLine<'a>,
-    headers: Headers,
-    body: Option<Body>,
+pub struct Request {
+    /// The request line of the request.
+    pub request_line: RequestLine,
+    /// The headers of the request.
+    pub headers: Headers,
+    /// The body of the request.
+    pub body: Option<Body>,
 }
 
-impl<'a> Request<'a> {
+impl Request {
+    /// Default request constructor.
+    pub fn default() -> Self {
```

Comment:
Instead of doing this you should implement Default for Request which is different from having a default function in the Request implementation.